File: archived/src/c/ch01/1.3/l0/exercises_starter.c
#include <stdio.h>
#include <stdint.h>
#include <stdbool.h>
#include "parity.h"
#include "divisibility.h"
#include "remainder.h"

const char *exercise_1(int64_t n) { return is_even(n) ? "even" : "odd"; }
bool        exercise_2(void)      { return is_divisible(91, 7); }
int64_t     exercise_3(void)      { int64_t q,r; div_identity(100,9,&q,&r); return r; }
int         exercise_4(void)      { return week_shift(5, 10); }     /* Sat + 10 -> Mon(1) */
int64_t     exercise_5(void)      { return powmod(2, 15, 10); }     /* last digit of 2^15 */

int main(void) {
    printf("Exercise 1 (42 parity): %s\n", exercise_1(42));
    printf("Exercise 2 (91 divisible by 7?): %s\n", exercise_2() ? "true" : "false");
    printf("Exercise 3 (100 %% 9): %lld\n", (long long)exercise_3());
    printf("Exercise 4 (Sat+10): %d\n", exercise_4());
    printf("Exercise 5 (last digit of 2^15): %lld\n", (long long)exercise_5());
    return 0;
}

File: archived/src/erlang/ch01/1.3/l0/src/exercises_starter.erl
-module(exercises_starter).
-export([exercise1/1, exercise2/0, exercise3/0, exercise4/0, exercise5/0, main/0]).

exercise1(N) -> if parity:is_even(N) -> <<"even">>; true -> <<"odd">> end.
exercise2()  -> divisibility:is_divisible(91,7).
exercise3()  -> element(2, remainder:div_identity(100,9)).   %% remainder
exercise4()  -> remainder:week_shift(5,10).                   %% Sat + 10 -> Mon(1)
exercise5()  -> remainder:powmod(2,15,10).                    %% last digit of 2^15

main() ->
    io:format("Exercise 1 (42 parity): ~s~n", [exercise1(42)]),
    io:format("Exercise 2 (91 divisible by 7?): ~p~n", [exercise2()]),
    io:format("Exercise 3 (100 % 9): ~p~n", [exercise3()]),
    io:format("Exercise 4 (Sat+10): ~p~n", [exercise4()]),
    io:format("Exercise 5 (last digit of 2^15): ~p~n", [exercise5()]).
